### PR TITLE
Use simplified version of ABCI++ 

### LINKED ISSUE #204: Upgrade to first ABCI++ celestia branch
Currently the app and core utilize the ABCI to communicate. This limits the possible solutions we can pursue for important features such as [#455](https://github.com/celestiaorg/celestia-core/issues/455), [#454](https://github.com/celestiaorg/celestia-core/issues/454), and [#77](https://github.com/celestiaorg/celestia-core/issues/77). While the current implementation of ABCI++ is not entirely stable, we should incorporate it so that we can begin to work on solutions to issues above. 

This will involve
- using the master branch (or whatever the abci++ branch ends up being) of celestia-core 
- Refactoring the `PreprocessTxs` method to `PrepareProposal`

### PR DESCRIPTION
**Pull request by evan-forbes**
## Description

This branch uses the new ABCI++ branches created in [#631](https://github.com/celestiaorg/celestia-core/pull/631) and [#63](https://github.com/celestiaorg/cosmos-sdk/pull/63)

closes #204 
blocked by [#631](https://github.com/celestiaorg/celestia-core/pull/631) and [#63](https://github.com/celestiaorg/cosmos-sdk/pull/63)
unblocks our wildest dreams (lol)

### CODE REVIEW

**Comment by adlerjohn** [comment]
Is this still blocked? If so, can we convert to draft?

File: go.mod [hunk 1]
```diff
@@ -124,8 +124,8 @@ require (
 
 replace (
 	github.com/99designs/keyring => github.com/cosmos/keyring v1.1.7-0.20210622111912-ef00f8ac3d76
-	github.com/cosmos/cosmos-sdk => github.com/celestiaorg/cosmos-sdk v0.44.1-celestia
+	github.com/cosmos/cosmos-sdk => github.com/celestiaorg/cosmos-sdk v0.44.2-0.20220216073333-2945fdd40acb
```
Comment (liamsi) [inline go.mod:127]: Please use a tagged version/release here once the sdk PR is merged.

**Review by liamsi — COMMENTED** [review]
> unblocks our wildest dreams (lol)

🤣 

Should use a tagged version once the SDK pr this is based on is merged.

**Comment by evan-forbes** [comment]
had to force push to fix conflicts

**Pull request merged**